pkg/retention/retention.go

```diff
@@ -416,3 +419,108 @@ func DeleteMetricsSegmentData(mmetaFile string, metricSegmentsToDelete map[strin
 		return
 	}
 }
+
+func doInodeBasedDeletion(ingestNodeDir string, maxInodeUsagePercent uint64, deletionWarningCounter int) {
+	var stat syscall.Statfs_t
+	dataPath := config.GetDataPath()
+
+	err := syscall.Statfs(filepath.Clean(dataPath), &stat)
+	if err != nil {
+		log.Errorf("doInodeBasedDeletion: Failed to get inode stats: %v", err)
+		return
+	}
+
+	totalInodes := stat.Files
+	freeInodes := stat.Ffree
+	usedInodes := totalInodes - freeInodes
+
+	if totalInodes == 0 {
+		log.Errorf("doInodeBasedDeletion: Invalid total inodes count")
+		return
+	}
+
+	currentUsagePercent := (usedInodes * 100) / totalInodes
+
+	log.Infof("doInodeBasedDeletion: Current inode usage: %v%%, Max allowed: %v%%, IngestNodeDir: %v",
```

Comment:
Maybe add the actual values here since that would be more helpful in debugging rather than just percentages